specifyweb/frontend/js_src/lib/components/DataModel/businessRuleDefs.ts

```diff
@@ -241,6 +241,23 @@ export const businessRuleDefs: MappedBusinessRuleDefs = {
   },
 
   CollectionObjectGroupJoin: {
+    customInit: (
+      CollectionObjectGroupJoin: SpecifyResource<CollectionObjectGroupJoin>
+    ): void => {
+      if (
+        CollectionObjectGroupJoin.get('childCog') ===
+          CollectionObjectGroupJoin.get('parentCog') &&
+        typeof CollectionObjectGroupJoin.get('childCog') === 'string' &&
+        typeof CollectionObjectGroupJoin.get('parentCog') === 'string'
+      ) {
+        setSaveBlockers(
+          CollectionObjectGroupJoin,
+          CollectionObjectGroupJoin.specifyTable.field.childCog,
+          ['Cog added to itself'],
+          'Cog to COG'
```

Comment:
I think the blocker message can be added to a localization file and the blocker key can be added to `businessRulesUtils.ts` with other keys:
https://github.com/specify/specify7/blob/6b071d7ce274d5f85d17974e4c8e368b96727f3f/specifyweb/frontend/js_src/lib/components/DataModel/businessRuleUtils.ts#L5-L7